src/js/samples/components/Sidebar/ManageLabels.js

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import { connect } from "react-redux";
+import { SidebarHeader, SideBarSection } from "../../../base";
+import { SmallSampleLabel } from "../Label";
+import { SampleSidebarMultiselectList } from "./List";
+import { SampleSidebarSelector } from "./Selector";
+import { Link } from "react-router-dom";
+import styled from "styled-components";
+import { getFontSize, fontWeight, getColor } from "../../../app/theme";
+import { map, mapValues, reduce, forEach, filter, union, reject, flatMap } from "lodash-es";
+import { editSample } from "../../actions";
+
+const SampleLabelsFooter = styled.div`
+    display: flex;
+    color: ${props => getColor({ theme: props.theme, color: "greyDarkest" })};
+    a {
+        margin-left: 5px;
+        font-size: ${getFontSize("md")};
+        font-weight: ${fontWeight.thick};
+    }
+`;
+
+const StyledSideBarSection = styled(SideBarSection)`
+    grid-row: 2;
+    align-self: start;
+`;
+
+const getSelectedSamples = (selected, documents) => {
+    return filter(documents, document => selected.includes(document.id));
+};
```

Comment:
This is not efficient as you are calling `includes` for every `document`. This will bog down with larger sizes of `selected` and is a good example for thinking in terms of big O notation.

There is probably a function in Lodash that will do this for you much more efficiently.

Reply:
Turns out there was an existing `getSelectedSamples` in `samples/selectors.js` with identical implementation. I removed my own and replaced the code in the existing one with a version that should run faster.